### DISCUSSION TITLE
Set fill height for custom bins

### DISCUSSION THREAD

**Discussion by unteins** [kennetek/gridfinity-rebuilt-openscad#11] [Ideas]
It would be nice to be able to control the height of the filled block independent from the height of the walls of the bins when making custom boxes.

Here's my specific use case:

I am making a custom bin to hold my screwdrivers. I want the screwdrivers to be easy to remove from the bin, but I ALSO want to be able to stack another bin on top of this bin. The best way to do that is to have the wall height be say 6 Z units and the interior top of the block that my custom cuts are removed from be say 2.5 Z units or 45% of the wall height. 

Right now I am attempting to achieve this with a difference, but there's no easy way to calculate the size and position of the cube I need, so it is a lot of trial and error. Because OpenSCAD can't determine the size of something you have to tell it, this would be best as an additional parameter that can be passed to the cut_move function. 

If you allowed the passing of two variables with a bottom and top height you could create stepped bins where one side of the cutout was lower than the other.

One other advantage is this could greatly reduce print times and filament consumption as the extra needed bulk of the bin wouldn't be required.

The first image is the custom bin I am working on (got measurements wrong) and the second is my attempt to chop out a portion of the upper part of the bin to make the screwdrivers easier to pull out.

![IMG_4841](https://user-images.githubusercontent.com/73155301/192007681-f552a6fa-de13-4fb1-8c93-49eb48ba5b1d.jpeg)
![Screen Shot 2022-09-23 at 9 23 00 AM](https://user-images.githubusercontent.com/73155301/192007558-abc84cb0-9634-4328-ad31-2f29b748e90e.png)

**Comment by kennetek** [comment]
Reviewing this made me realize an error with `cut_move`. It does not behave the way I expected it to, which is why you have been having such a hard time with depth. It does not translate the objects on the Z-axis, which I thought it did. I need to figure out the best way to fix that, thanks for making me realize it. 

As for the internal height, I will add another option for that to make it easier, but a quick fix is to go to `gridfinity-rebuilt-base.scad` and edit the `module gridfinityCustom`. Where it says `block_bottom()`, replace the value with your desired height, in millimeters. That's the internal solid block height, measured from the top of the bases.

**Comment by kennetek** [comment]
Little late, but there is now a parameter to override the internal height, added in a recent commit f74618f86e924bd7dd03b743c1b6395178af7404. Basically, the XY plane is translated to the top of the internal bin area, and `cut_move` locates it's XY position based on what section of the grid the pattern should be in.
